app/repos/search.js

```diff
@@ -1,10 +1,13 @@
 const sequelize = require('../config/db')
+const { dbFindByPk } = require('../lib/db-functions')
 
-const addToSearchIndex = async (person, dog, transaction) => {
+const addToSearchIndex = async (person, dogId, transaction) => {
   if (!transaction) {
-    return sequelize.transaction((t) => addToSearchIndex(person, dog, t))
+    return sequelize.transaction((t) => addToSearchIndex(person, dogId, t))
   }
 
+  const dog = await dbFindByPk(sequelize.models.dog, dogId, { raw: true, nest: true, transaction })
```

Comment:
Could we not add a `getDog` function to the dog repo? Or have you made this separate as a one off here as `getDog` will be using the full index number rather than the id?